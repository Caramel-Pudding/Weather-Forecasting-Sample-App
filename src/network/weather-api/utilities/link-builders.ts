import { apiKeyStoredInACompleteSecurity } from "@/consts/mocked";
import { weatherAPIBasicRoute } from "../consts/routes";

export const buildWeatherAPIRequestRoute = (query: string): string =>
  `${weatherAPIBasicRoute}/data/2.5/forecast?q=${query}&appid=${apiKeyStoredInACompleteSecurity}`;
