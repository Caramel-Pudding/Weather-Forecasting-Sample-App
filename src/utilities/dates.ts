import { DayOfTheWeek, Month } from "@/consts/dates";

export const convertUnixTimestampToDate = (unixTimestamp: number): Date =>
  new Date(unixTimestamp * 1000);

export const getDayAsTextFromTimestamp = (timestamp: Date): string =>
  DayOfTheWeek[timestamp.getDay()];

export const getMonthAsTextFromTimestamp = (timestamp: Date): string =>
  Month[timestamp.getMonth()];

export const getTimeAsTextFromTimestamp = (timestamp: Date): string => {
  return `${timestamp.getHours()}:00`;
};
