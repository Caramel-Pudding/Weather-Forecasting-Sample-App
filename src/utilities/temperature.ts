export const convertKelvinToCelsius = (kelvin: number): number =>
  Math.round(kelvin - 273.15);
