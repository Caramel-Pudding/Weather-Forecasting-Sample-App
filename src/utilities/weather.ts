import { WeatherListItem } from "@/types/weather";

export const getWeatherListForToday = (
  weatherList: WeatherListItem[]
): WeatherListItem[] => {
  if (!weatherList || weatherList.length === 0) {
    return [];
  }
  const todayDate = new Date(weatherList[0].dt_txt).getDate();
  return weatherList.filter(
    (listItem) => new Date(listItem.dt_txt).getDate() === todayDate
  );
};

export const getTimeStringFromWeatherTimestamp = (
  weatherTimestamp: string
): string => {
  const timestamp = new Date(weatherTimestamp);
  return `${timestamp.getHours()}:00`;
};
