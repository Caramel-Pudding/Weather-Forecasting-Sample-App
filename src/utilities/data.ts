import { WeatherListItem } from "@/types/weather";
import { convertUnixTimestampToDate } from "./dates";

export const getWeatherListForToday = (
  weatherList: WeatherListItem[]
): WeatherListItem[] => {
  if (!weatherList || weatherList.length === 0) {
    return [];
  }

  const todayDate = convertUnixTimestampToDate(weatherList[0].dt).getDate();
  return weatherList.filter(
    (listItem) =>
      convertUnixTimestampToDate(listItem.dt).getDate() === todayDate
  );
};
