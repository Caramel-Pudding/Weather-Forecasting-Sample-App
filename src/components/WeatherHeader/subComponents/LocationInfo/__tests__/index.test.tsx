import React from "react";
import { render, cleanup } from "@/tests/test-utils";

import { convertUnixTimestampToDate } from "@/utilities/dates";
import { weatherDataStub } from "@/tests/stubs";
import { LocationInfo } from "..";

describe("LocationInfo Component", () => {
  afterEach(cleanup);

  it("Should render properly if there's proper data", () => {
    // * #TEST: ARRANGE
    const { container, getByText } = render(
      <LocationInfo
        cityName={weatherDataStub.city.name}
        currentDate={convertUnixTimestampToDate(weatherDataStub.list[0].dt)}
      />,
      {}
    );
    const city = getByText("Altstadt");
    const day = getByText("Friday");
    const date = getByText("16. February");
    // * #TEST: ASSERT
    expect(city).toBeInTheDocument();
    expect(day).toBeInTheDocument();
    expect(date).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
