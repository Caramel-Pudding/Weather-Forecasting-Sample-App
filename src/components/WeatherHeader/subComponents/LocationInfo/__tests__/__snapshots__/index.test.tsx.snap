// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationInfo Component Should render properly if there's proper data 1`] = `
<div>
  <article
    class="container"
  >
    <section>
      <div
        class="additionalText"
      >
        Altstadt
      </div>
      <section
        class="mainText date"
      >
        <div>
          Friday
        </div>
        <div>
          16. February
        </div>
      </section>
    </section>
  </article>
</div>
`;
